src/main/drivers/serial_pinconfig.c

```diff
@@ -0,0 +1,275 @@
+#endif
+#endif
+
+#ifdef USE_UART5
+#ifndef UART5_TX_PIN
+#define UART5_TX_PIN        PC12
+#endif
+#ifndef UART5_RX_PIN
+#define UART5_RX_PIN        PD2
+#endif
+#endif
+#endif // STM32F3
+
+// Default pin (NONE).
+
+#ifdef USE_UART1
+# if !defined(UART1_RX_PIN)
+#  define UART1_RX_PIN NONE
+# endif
+# if !defined(UART1_TX_PIN)
+#  define UART1_TX_PIN NONE
+# endif
+# if !defined(INVERTER_PIN_UART1)
+#  define INVERTER_PIN_UART1 NONE
+# endif
+#endif
+
+#ifdef USE_UART2
+# if !defined(UART2_RX_PIN)
+#  define UART2_RX_PIN NONE
+# endif
+# if !defined(UART2_TX_PIN)
+#  define UART2_TX_PIN NONE
+# endif
+# if !defined(INVERTER_PIN_UART2)
+#  define INVERTER_PIN_UART2 NONE
+# endif
+#endif
+
+#ifdef USE_UART3
+# if !defined(UART3_RX_PIN)
+#  define UART3_RX_PIN NONE
+# endif
+# if !defined(UART3_TX_PIN)
+#  define UART3_TX_PIN NONE
+# endif
+# if !defined(INVERTER_PIN_UART3)
+#  define INVERTER_PIN_UART3 NONE
+# endif
+#endif
+
+#ifdef USE_UART4
+# if !defined(UART4_RX_PIN)
+#  define UART4_RX_PIN NONE
+# endif
+# if !defined(UART4_TX_PIN)
+#  define UART4_TX_PIN NONE
+# endif
+# if !defined(INVERTER_PIN_UART4)
+#  define INVERTER_PIN_UART4 NONE
+# endif
+#endif
+
+#ifdef USE_UART5
+# if !defined(UART5_RX_PIN)
+#  define UART5_RX_PIN NONE
+# endif
+# if !defined(UART5_TX_PIN)
+#  define UART5_TX_PIN NONE
+# endif
+# if !defined(INVERTER_PIN_UART5)
+#  define INVERTER_PIN_UART5 NONE
+# endif
+#endif
+
+#ifdef USE_UART6
+# if !defined(UART6_RX_PIN)
+#  define UART6_RX_PIN NONE
+# endif
+# if !defined(UART6_TX_PIN)
+#  define UART6_TX_PIN NONE
+# endif
+# if !defined(INVERTER_PIN_UART6)
+#  define INVERTER_PIN_UART6 NONE
+# endif
+#endif
+
+#ifdef USE_UART7
+# if !defined(UART7_RX_PIN)
+#  define UART7_RX_PIN NONE
+# endif
+# if !defined(UART7_TX_PIN)
+#  define UART7_TX_PIN NONE
+# endif
+# if !defined(INVERTER_PIN_UART7)
+#  define INVERTER_PIN_UART7 NONE
+# endif
+#endif
+
+#ifdef USE_UART8
+# if !defined(UART8_RX_PIN)
+#  define UART8_RX_PIN NONE
+# endif
+# if !defined(UART8_TX_PIN)
+#  define UART8_TX_PIN NONE
+# endif
+# if !defined(INVERTER_PIN_UART8)
+#  define INVERTER_PIN_UART8 NONE
+# endif
+#endif
+
+#ifdef USE_SOFTSERIAL1
+# if !defined(SOFTSERIAL1_RX_PIN)
+#  define SOFTSERIAL1_RX_PIN NONE
+# endif
+# if !defined(SOFTSERIAL1_TX_PIN)
+#  define SOFTSERIAL1_TX_PIN NONE
+# endif
+#endif
+
+#ifdef USE_SOFTSERIAL2
+# if !defined(SOFTSERIAL2_RX_PIN)
+#  define SOFTSERIAL2_RX_PIN NONE
+# endif
+# if !defined(SOFTSERIAL2_TX_PIN)
+#  define SOFTSERIAL2_TX_PIN NONE
+# endif
+#endif
+
+// XXX Should these be moved to drivers/serial.c?
+
+#if defined(USE_UART) || defined(USE_SOFTSERIAL1) || defined(USE_SOFTSERIAL1)
+typedef struct serialDefaultPin_s {
+    serialPortIdentifier_e ident;
+    ioTag_t rxIO, txIO, inverterIO;
+} serialDefaultPin_t;
+
+static const serialDefaultPin_t serialDefaultPin[] = {
+#ifdef USE_UART1
+    { SERIAL_PORT_USART1, DEFIO_TAG_E(UART1_RX_PIN), DEFIO_TAG_E(UART1_TX_PIN), DEFIO_TAG_E(INVERTER_PIN_UART1) },
```

Comment:
Not sure about `DEFIO_TAG_E`. It will use NONE when target does not contain corresponding pin, this may hide configuration problem (`target.h` trying to use pin that is not available).